File: C++/191.cpp
__________________________________________________________________________________________________
sample 4 ms submission
class Solution {
public:
    int hammingWeight(uint32_t n) {
        int count=0;
        while(n!=0)
        {
            if((n&1) ==1)
                count+=1;
            n=n>>1;
        }
        return count;
    }
};
__________________________________________________________________________________________________
sample 7852 kb submission
class Solution {
public:
    int hammingWeight(uint32_t n) {
            int res = 0;
    while(n)
    {
        n &= n - 1;
        res++;
    }
    return res;
    }
};
__________________________________________________________________________________________________

File: Java/191.java
__________________________________________________________________________________________________
sample 0 ms submission
public class Solution {
    // you need to treat n as an unsigned value
    public int hammingWeight(int n) {
       int count =0;
        while(n!=0){
           count+=(n & 0x1);
            n = n>>>1;
            
       } 
        return count;
    }
}
__________________________________________________________________________________________________
sample 31760 kb submission
public class Solution {
    // you need to treat n as an unsigned value
    public int hammingWeight(int n) {
        int count = 0;
        while(n!=0){
            if((n&1) == 1){count++;}
            n = n>>>1; //这里的无符号位移
        }
        return count;
    }
}
__________________________________________________________________________________________________
